### PR TITLE
Feature/macgills/#1292 language activity

### LINKED ISSUE #1292: Use LanguageActivity
**Is your feature request related to a problem? Please describe.**
Due to merge problems with 2.5->3.0 LanguageActivity is no longer used but instead the older LanguageSelectDialog

**Describe the solution you'd like**
Revert to the 3.0 LanguageActivity

### PR DESCRIPTION
**Pull request by macgills**
Fixes #1292

Changes: Full rewrite to kotlin of LanguageActivity

### CODE REVIEW

**Comment by abdulwd** [comment]
When I click the globe icon to open the activity the app crashes. Here's the stack trace:
```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.kiwix.kiwixmobile/org.kiwix.kiwixmobile.language.LanguageActivity}: java.lang.IllegalStateException: Required view 'text_input_password_toggle' with ID 2131296595 for field 'toolbar' was not found. If this view is optional add '@Nullable' (fields) or '@Optional' (methods) annotation.
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3260)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3396)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2009)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7319)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:934)
     Caused by: java.lang.IllegalStateException: Required view 'text_input_password_toggle' with ID 2131296595 for field 'toolbar' was not found. If this view is optional add '@Nullable' (fields) or '@Optional' (methods) annotation.
        at butterknife.internal.Utils.findRequiredView(Utils.java:88)
        at butterknife.internal.Utils.findRequiredViewAsType(Utils.java:100)
        at org.kiwix.kiwixmobile.language.LanguageActivity_ViewBinding.<init>(LanguageActivity_ViewBinding.java:27)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at butterknife.ButterKnife.bind(ButterKnife.java:170)
        at butterknife.ButterKnife.bind(ButterKnife.java:99)
        at org.kiwix.kiwixmobile.base.BaseActivity.setContentView(BaseActivity.java:61)
        at org.kiwix.kiwixmobile.language.LanguageActivity.onCreate(LanguageActivity.kt:47)
        at android.app.Activity.performCreate(Activity.java:7783)
        at android.app.Activity.performCreate(Activity.java:7772)
```

**Comment by macgills** [comment]
@abdulwd I think you must not have the right version, LanguageActivity does not find views with Butterknife. It is possible that you need to clean the project before building as the auto generated butterknife files might be lingering around from a previous build?

**Comment by abdulwd** [comment]
> @abdulwd I think you must not have the right version, LanguageActivity does not find views with Butterknife. It is possible that you need to clean the project before building as the auto generated butterknife files might be lingering around from a previous build?

Indeed! I did not check the code. Android extensions do make things simpler.

**Comment by macgills** [comment]
@abdulwd extensions are fantastic, though I spent so many years snake_casing_my_view_ids and now they look so ugly in kotlin. A hard habit to break

File: app/src/main/java/org/kiwix/kiwixmobile/language/adapter/LanguageListViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    val clickAction: (LanguageItem) -> Unit
+  ) : LanguageListViewHolder<LanguageItem>(view) {
+    override fun bind(item: LanguageItem) {
+      val language = item.language
+      item_language_name.text = language.language
+      item_language_localized_name.text = language.languageLocalized
+      item_language_books_count.text = containerView.resources.getQuantityString(
+          R.plurals.books_count, language.occurencesOfLanguage, language.occurencesOfLanguage
+      )
+      item_language_checkbox.isChecked = language.active;
```
[the remark this answers is not in the document]
Comment (macgills) [reply app/src/main/java/org/kiwix/kiwixmobile/language/adapter/LanguageListViewHolder.kt]: done

File: app/src/main/java/org/kiwix/kiwixmobile/language/viewmodel/State.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package org.kiwix.kiwixmobile.language.viewmodel
+
+import org.kiwix.kiwixmobile.language.adapter.LanguageListItem
+import org.kiwix.kiwixmobile.language.adapter.LanguageListItem.HeaderItem
+import org.kiwix.kiwixmobile.language.adapter.LanguageListItem.LanguageItem
+import org.kiwix.kiwixmobile.zim_manager.Language
+
+sealed class State() {
```
[the remark this answers is not in the document]
Comment (macgills) [reply app/src/main/java/org/kiwix/kiwixmobile/language/viewmodel/State.kt]: done

**Comment by macgills** [comment]
@abdulwd You raise an interesting point around these inspection (technically not lint) failures, I should set up a pre commit hook to run detekt or at least integrate detekt with the CI

**Pull request merged**